wolftpm/tpm2_wrap.h

```diff
@@ -118,214 +118,1699 @@ typedef struct WOLFTPM2_CAPS {
+*/
 WOLFTPM_API int wolfTPM2_ECDHEGenKey(WOLFTPM2_DEV* dev, WOLFTPM2_KEY* ecdhKey,
     int curve_id);
+
+/*!
+    \ingroup wolfTPM2_Wrappers
+    \brief Computes Z (shared secret) using pubPoint and counter (2 phase method)
+    Note: The counter, array ID, can only be used one time
+
+    \return TPM_RC_SUCCESS: successful
+    \return TPM_RC_FAILURE: generic failure (check TPM IO and TPM return code)
+    \return BAD_FUNC_ARG: check the provided arguments
+
+    \param dev pointer to a TPM2_DEV struct
+    \param parentKey pointer to a structure of WOLFTPM2_KEY type, containing a valid TPM handle of a primary key
+    \param ecdhKey pointer to a structure of WOLFTPM2_KEY type, containing a valid TPM handle
+    \param out pointer to a byte buffer, to store the computed shared secret
+    \param outSz integer value, specifying the size of the shared secret, in bytes
+
+    \sa wolfTPM2_ECDHEGenKey
+    \sa wolfTPM2_ECDHGen
+    \sa wolfTPM2_ECDHGenKey
+    \sa wolfTPM2_ECDHGenZ
+*/
 WOLFTPM_API int wolfTPM2_ECDHEGenZ(WOLFTPM2_DEV* dev, WOLFTPM2_KEY* parentKey,
     WOLFTPM2_KEY* ecdhKey, const TPM2B_ECC_POINT* pubPoint,
     byte* out, int* outSz);
 
+/*!
+    \ingroup wolfTPM2_Wrappers
+    \brief Perform RSA encryption using a TPM 2.0 key
+
+    \return TPM_RC_SUCCESS: successful
+    \return TPM_RC_FAILURE: generic failure (check TPM IO and TPM return code)
+    \return BAD_FUNC_ARG: check the provided arguments
+
+    \param dev pointer to a TPM2_DEV struct
+    \param key pointer to a struct of WOLFTPM2_KEY type, holding a TPM key material
+    \param padScheme integer value of TPM_ALG_ID type, specifying the padding scheme
+    \param msg pointer to a byte buffer, containing the arbitrary data for encryption
+    \param msgSz integer value, specifying the size of the arbitrary data buffer
+    \param out pointer to a byte buffer, where the encrypted data will be stored
+    \param outSz integer value, specifying the size of the encrypted data buffer
+
+    \sa wolfTPM2_RsaDecrypt
+*/
 WOLFTPM_API int wolfTPM2_RsaEncrypt(WOLFTPM2_DEV* dev, WOLFTPM2_KEY* key,
     TPM_ALG_ID padScheme, const byte* msg, int msgSz, byte* out, int* outSz);
+
+/*!
+    \ingroup wolfTPM2_Wrappers
+    \brief Perform RSA decryption using a TPM 2.0 key
+
+    \return TPM_RC_SUCCESS: successful
+    \return TPM_RC_FAILURE: generic failure (check TPM IO and TPM return code)
+    \return BAD_FUNC_ARG: check the provided arguments
+
+    \param dev pointer to a TPM2_DEV struct
+    \param key pointer to a struct of WOLFTPM2_KEY type, holding a TPM key material
+    \param padScheme integer value of TPM_ALG_ID type, specifying the padding scheme
+    \param in pointer to a byte buffer, containing the encrypted data
+    \param inSz integer value, specifying the size of the encrypted data buffer
+    \param msg pointer to a byte buffer, containing the decrypted data
+    \param msgSz integer value, specifying the size of the encrypted data buffer
```

Comment:
I think this is a pointer used for max buffer size on input and set to actual size used on success. If so, could be clearer